plugwise/helper.py

```diff
@@ -1069,24 +1117,60 @@ def _schedules(self, location: str) -> tuple[list[str], str, str | None]:
 
         tag = "zone_preset_based_on_time_and_presence_with_override"
         if not (rule_ids := self._rule_ids_by_tag(tag, location)):
-            return available, selected, None
+            return available, selected, schedule_temperatures, None
 
-        schedules: list[str] = []
+        schedules: dict[str, dict[str, list[float]]] = {}
         for rule_id, loc_id in rule_ids.items():
             name = self._domain_objects.find(f'./rule[@id="{rule_id}"]/name').text
-            available.append(name)
-            if location == loc_id:
-                selected = name
-                self._last_active[location] = selected
-            schedules.append(name)
+            schedule: dict[str, list[float]] = {}
+            locator = f'./rule[@id="{rule_id}"]/directives'
+            directives = self._domain_objects.find(locator)
+            count = 0
+            for directive in directives:
+                count += 1
+                # Only process the schedule in detail for Anna with cooling
+                if self._anna_cooling_present:
```

Comment:
Why loop if it wouldn't be anna cooling? Suggest to first do the if cooling then loop?

Reply:
Because then it's not needed/not useful. That's the simplification I implemented some time ago.

Reply:
What I meant is coding-wise that the loop just counts and executes if cooling present, as thats the only positive flow. If that assumption is right it could be 'if colling_present: loop' instead?

Reply:
I want to count always, to find and be able to ignore any empty schedules.
And count and process for an Anna with cooling. Because the `anna_heatpump`-based datasets also have an empty schedule.

Reply:
I'm referring to reversing L1132 and L1129, not the L1123 one?

Reply:
Yes I understood. But then the question is, what to do when a schedule is empty?
In v0.18.5 we allow them. I'm not sure that's the correct strategy...

Reply:
I can't create an empty schedule via the website. Or via the App.
This must have been possible in the past because there is an empty schedule in our userdata.

Reply:
Update: datasets fixed, moved guard before looping.

Reply:
We should fix the dataset, my 'Test' was empty as well in the past but it isn't possible to create such schema's any longer.